### ISSUE TITLE
Missing `aggreagate` kwarg for `esm_datastore.to_dataset_dict`

### ISSUE THREAD

**Issue by jbusecke** [intake/intake-esm#256]
When passing `aggregate=False` to `.to_dataset_dict` I get the following error:

```
TypeError                                 Traceback (most recent call last)
<ipython-input-8-aace3fd1d074> in <module>
      2 col = intake.open_esm_datastore(url)
      3 cat = col.search(variable_id='o2', source_id='CanESM5', experiment_id='historical')
----> 4 dd = cat.to_dataset_dict(zarr_kwargs={'consolidated':True}, aggregate=False)

TypeError: to_dataset_dict() got an unexpected keyword argument 'aggregate'
```
It seems as if that option was deprecated, but not removed from the docstring. Is there another way to recover the old behavior?

**Comment by andersy005** [comment]
Good catch. I meant to address this issue a while ago, but I ended up forgetting. 

> Is there another way to recover the old behavior?

Currently, there is no easy way to recover the old behavior without some manipulation of `col.esmcol_data` attribute. What is the output of `col.esmcol_data`?

**Comment by jbusecke** [comment]
I am getting: 
```
{'esmcat_version': '0.1.0',
 'id': 'glade-cmip6',
 'description': 'This is an ESM collection for CMIP6 data accessible on the Princeton University disk storage system in /scratch/gpfs/GEOCLIM/synda/data/CMIP6',
 'catalog_file': '/home/jbusecke/projects/cmip_data_management_princeton/catalogs/tigressdata-cmip6.csv.gz',
 'attributes': [{'column_name': 'activity_id',
   'vocabulary': 'https://raw.githubusercontent.com/WCRP-CMIP/CMIP6_CVs/master/CMIP6_activity_id.json'},
  {'column_name': 'source_id',
   'vocabulary': 'https://raw.githubusercontent.com/WCRP-CMIP/CMIP6_CVs/master/CMIP6_source_id.json'},
  {'column_name': 'institution_id',
   'vocabulary': 'https://raw.githubusercontent.com/WCRP-CMIP/CMIP6_CVs/master/CMIP6_institution_id.json'},
  {'column_name': 'experiment_id',
   'vocabulary': 'https://raw.githubusercontent.com/WCRP-CMIP/CMIP6_CVs/master/CMIP6_experiment_id.json'},
  {'column_name': 'member_id', 'vocabulary': ''},
  {'column_name': 'table_id',
   'vocabulary': 'https://raw.githubusercontent.com/WCRP-CMIP/CMIP6_CVs/master/CMIP6_table_id.json'},
  {'column_name': 'variable_id', 'vocabulary': ''},
  {'column_name': 'grid_label',
   'vocabulary': 'https://raw.githubusercontent.com/WCRP-CMIP/CMIP6_CVs/master/CMIP6_grid_label.json'},
  {'column_name': 'version', 'vocabulary': ''},
  {'column_name': 'dcpp_start_year', 'vocabulary': ''}],
 'assets': {'column_name': 'path', 'format': 'netcdf'},
 'aggregation_control': {'variable_column_name': 'variable_id',
  'groupby_attrs': ['activity_id',
   'institution_id',
   'source_id',
   'experiment_id',
   'table_id',
   'grid_label'],
  'aggregations': [{'type': 'union', 'attribute_name': 'variable_id'},
   {'type': 'join_existing',
    'attribute_name': 'time_range',
    'options': {'dim': 'time', 'coords': 'minimal', 'compat': 'override'}},
   {'type': 'join_new',
    'attribute_name': 'member_id',
    'options': {'coords': 'minimal', 'compat': 'override'}},
   {'type': 'join_new',
    'attribute_name': 'dcpp_init_year',
    'options': {'coords': 'minimal', 'compat': 'override'}}]}}
```

**Comment by andersy005** [comment]
@jbusecke, I addressed this issue in #256. So, if you need this feature, try the master branch. I plan to cut a new release in the next few days.



```python
In [2]: col = intake.open_esm_datastore("./tests/sample-collections/catalog-di
   ...: ct-records.json")

In [3]: col
Out[3]: <aws-cesm1-le catalog with 1 dataset(s) from 5 asset(s)>

In [5]: col.df.head()
Out[5]:
  component frequency experiment variable                                               path
0       atm     daily        20C     FLNS  s3://ncar-cesm-lens/atm/daily/cesmLE-20C-FLNS....
1       atm     daily        20C    FLNSC  s3://ncar-cesm-lens/atm/daily/cesmLE-20C-FLNSC...
2       atm     daily        20C     FLUT  s3://ncar-cesm-lens/atm/daily/cesmLE-20C-FLUT....
3       atm     daily        20C     FSNS  s3://ncar-cesm-lens/atm/daily/cesmLE-20C-FSNS....
4       atm     daily        20C    FSNSC  s3://ncar-cesm-lens/atm/daily/cesmLE-20C-FSNSC...

In [6]: ds = col.to_dataset_dict(storage_options={'anon': True})

--> The keys in the returned dictionary of datasets are constructed as follows:
	'component.experiment.frequency'
 |███████████████████████████████████████████████████████████████████████████| 100.00% [1/1 00:00<00:00]
In [7]: len(ds)
Out[7]: 1

In [8]: ds.keys()
Out[8]: dict_keys(['atm.20C.daily'])

In [9]: ds = col.to_dataset_dict(storage_options={'anon': True}, aggregate=False) #no aggregation

--> The keys in the returned dictionary of datasets are constructed as follows:
	'component.frequency.experiment.variable.path'
 |███████████████████████████████████████████████████████████████████████████| 100.00% [5/5 00:00<00:00]
In [10]: len(ds)
Out[10]: 5

In [11]: [key for key in ds.keys()]
Out[11]:
['atm.daily.20C.FSNS.s3://ncar-cesm-lens/atm/daily/cesmLE-20C-FSNS.zarr',
 'atm.daily.20C.FLNS.s3://ncar-cesm-lens/atm/daily/cesmLE-20C-FLNS.zarr',
 'atm.daily.20C.FLNSC.s3://ncar-cesm-lens/atm/daily/cesmLE-20C-FLNSC.zarr',
 'atm.daily.20C.FSNSC.s3://ncar-cesm-lens/atm/daily/cesmLE-20C-FSNSC.zarr',
 'atm.daily.20C.FLUT.s3://ncar-cesm-lens/atm/daily/cesmLE-20C-FLUT.zarr']
```
